exchanges/order/order_test.go

```diff
@@ -310,42 +309,22 @@ func TestInferCostsAndTimes(t *testing.T) {
 	t.Parallel()
 	var detail Detail
 	detail.InferCostsAndTimes()
-	if detail.Amount != detail.ExecutedAmount+detail.RemainingAmount {
-		t.Errorf(
-			"Order detail amounts not equals. Expected 0, received %f",
-			detail.Amount-(detail.ExecutedAmount+detail.RemainingAmount),
-		)
-	}
+	assert.Equal(t, detail.Amount, detail.ExecutedAmount+detail.RemainingAmount)
 
 	detail.CloseTime = time.Now()
 	detail.InferCostsAndTimes()
-	if detail.LastUpdated != detail.CloseTime {
-		t.Errorf(
-			"Order last updated not equals close time. Expected %s, received %s",
-			detail.CloseTime,
-			detail.LastUpdated,
-		)
-	}
+	assert.Equal(t, detail.CloseTime, detail.LastUpdated, "Order last updated not equals close time")
 
 	detail.Amount = 1
 	detail.ExecutedAmount = 1
 	detail.InferCostsAndTimes()
-	if detail.AverageExecutedPrice != 0 {
-		t.Errorf(
-			"Unexpected AverageExecutedPrice. Expected 0, received %f",
-			detail.AverageExecutedPrice,
-		)
-	}
+	assert.Zero(t, detail.AverageExecutedPrice, "Unexpected AverageExecutedPrice")
```

Comment:
🦍 Huge nitpick, but prefer assertion statements over error statements in tests.
```suggestion
	assert.Zero(t, detail.AverageExecutedPrice, "InferCostsAndTimes should set AverageExecutedPrice correctly")
```